### DISCUSSION TITLE
How to not override JWT session/extensions when using refresh token webhook?

### DISCUSSION THREAD

**Discussion by alekbarszczewski** [ory/hydra#3081] [Q&A] [answered]
I am adding custom session extensions to access token when accepting consent request:

```csharp
var request = new HydraAcceptConsentRequest(
    grantScope: scope.ToList(),
    session: new HydraConsentRequestSession(accessToken: new Dictionary<string, object>()
    {
        { "sid", sessionId },
        { "mfa", mfaVerified }
    })
);

var result = await _adminApi.AcceptConsentRequestAsync(consentChallenge, request);
```

And JWT token looks like this:

```json
{
  "sub": "xxx",
  ...
  "ext": { // <--- extensions
    "mfa": true,
    "sid": "14628d10-8c63-4c7f-8a9c-3e37318ce7d6"
  },
  ...
}
```

For some reasons I also use refresh token webhook (as described here https://www.ory.sh/docs/hydra/guides/claims-at-refresh). The problem is that I don't want to override JWT extensions. I tried returning empty json from refresh token webhook, but it removes "ext" field from JWT (new refreshed access token does not have this field at all). The problem is that current value (before refresh) of "ext" JWT field is not passed in any way to refresh token webhook and therefore I can't set those values again...

Any idea how to make refresh token webhook to no override "ext" field in JWT?

**Comment by aeneasr** [comment, marked as answer]
It looks like this needs code changes to work and include the previous extensions :) PRs welcomed!
